data/tokens/src/main/kotlin/com/tangem/data/tokens/repository/DefaultNetworksRepository.kt

```diff
@@ -107,11 +110,22 @@ internal class DefaultNetworksRepository(
         val userWallet = requireNotNull(userWalletsStore.getSyncOrNull(userWalletId)) {
             "Unable to find user wallet with provided ID: $userWalletId"
         }
-        val response = requireNotNull(userTokensStore.getSyncOrNull(userWalletId)) {
-            "Unable to find tokens response for user wallet with provided ID: $userWalletId"
-        }
 
-        return responseCurrenciesFactory.createCurrencies(response, userWallet.scanResponse.card)
+        return if (userWallet.isMultiCurrency) {
+            val response = requireNotNull(userTokensStore.getSyncOrNull(userWalletId)) {
+                "Unable to find tokens response for user wallet with provided ID: $userWalletId"
+            }
+
+            responseCurrenciesFactory.createCurrencies(response, userWallet.scanResponse.card)
+        } else {
+            val currency = cardCurrenciesFactory.createPrimaryCurrencyForSingleCurrencyCard(userWallet.scanResponse)
+
+            if (currency is CryptoCurrency.Token) {
+                listOf(currency)
+            } else {
+                emptyList()
+            }
```

Comment:
Вот тут вот так нужно

```suggestion
            listOf(currency)
```